app/models/user.rb

```diff
@@ -110,13 +110,11 @@ class User < ActiveRecord::Base
   scope :admins, -> { where('admin = ? OR role = ?',true,'admin').order('name asc') }
   scope :agents, -> { where('admin = ? OR role = ? OR role = ?',true,'admin','agent').order('name asc') }
   scope :team, -> { where('admin = ? OR role = ? OR role = ? OR role = ?',true,'admin','agent','editor').order('name asc') }
-  scope :active, -> { where('active = ?', true)}
-  scope :by_role, -> (role) { where('role = ?', role) }
+  scope :active, -> { where(active: true)}
+  scope :by_role, -> (role) { where(role: role) }
 
   def set_role_on_invitation_accept
-    if self.role.nil?
-      self.role = "agent"
-    end
+    self.role = self.role.presence || "agent"
```

Comment:
`self.role ||= "agent"`, is even cleaner imo.

Reply:
the reason to use `self.role.presence` as it will also exclude blank string.

     "" || "agent"
     => ""

Reply:
In theory the roles are predefined and should only ever be one of `[admin, agent, editor, user]`. 

Although I don't see any enforcement of this within the model.
Maybe that should be looked into, good point on the blank string case though.

Reply:
I would leave it unrestricted in case someone wants to add their own values.